src/mapgen/mg_ore.cpp

```diff
@@ -371,17 +371,19 @@ void OreVein::generate(MMVManip *vm, int mapseed, u32 blockseed,
 	PcgRandom pr(blockseed + 520);
 	MapNode n_ore(c_ore, 0, ore_param2);
 
-	u32 sizex = (nmax.X - nmin.X + 1);
-
-	if (!noise) {
-		int sx = nmax.X - nmin.X + 1;
-		int sy = nmax.Y - nmin.Y + 1;
-		int sz = nmax.Z - nmin.Z + 1;
-		noise  = new Noise(&np, mapseed, sx, sy, sz);
-		noise2 = new Noise(&np, mapseed + 436, sx, sy, sz);
+	int sizex = nmax.X - nmin.X + 1;
+	int sizey = nmax.Y - nmin.Y + 1;
+	// Because this ore uses 3D noise the perlinmap Y size can be different in
+	// different mapchunks due to ore Y limits. So recreate the noise objects
+	// if Y size has changed.
+	if (!noise || sizey != sizey_prev) {
+		int sizez = nmax.Z - nmin.Z + 1;
+		noise  = new Noise(&np, mapseed, sizex, sizey, sizez);
+		noise2 = new Noise(&np, mapseed + 436, sizex, sizey, sizez);
```

Comment:
Both noises need deletion when they were defined previously (non-nullptr).

Reply:
That's the OreVein destructor, which is only called when this class here is deleted. What we need is to destruct and delete/free the `Noise` class.

Reply:
I have no idea what is right to do. Also, such changes are needed for all ore types so should be a separate PR, will merge this bugfix first.

Reply:
Done. well spotted, deletion is needed for this ore type only because the noises are created multiple times.